### PR TITLE
APPEALS-42631- Create Rspec for Parser with sample payload method, and add additional parser methods 

### PR DESCRIPTION
**Pull request by Jruuuu**
## Resolves [APPEALS-42631](https://jira.devops.va.gov/browse/APPEALS-42631)

# Description
As a developer, I need to validate the correct working of the parser using a detailed unit rspec test that tests each parser method. This will be done by using a sample payload file saved along side the parser class. In this story we added an example_json to mock the payload response from appeals-consumer called decision_review_created_example.json. Additionally, added methods for Intake, Claimant,  Claim Review, and load_example in order to run our Rspec unit test.

`Events::DecisionReviewCreated::DecisionReviewCreatedParser.example_response` this method reads the decision_review_created_example.json

`Events::DecisionReviewCreated::DecisionReviewCreatedParser.load_example` this method invokes a new instance of the parser class with a sample_header & example_response to confirm the proper functionality of the `DecisionReviewCreatedParser`, JSON parser. 

## Acceptance Criteria
- [x] create a sample paylod file called devision_review_created_example.json at the same level as the parser
- [x] add a class method to the parser that will load the sample response called self.example_response
- [x] add a class method called ... self.load_example ... to the parser that will parse the JSON in the example_respose and return a new instance
- [x] of the parser .. new(JSON.parse(example_response))
- [x] add parser methods for claimant, claim_review, intake from payload 
- [x] Build out Rspec that test each method in the parser that it parsed the correct value loaded using the load_example method

## Testing Plan

1. [Jira Issue/Test Plan - APPEALS-43400](https://jira.devops.va.gov/browse/APPEALS-43400)
2. [X-Ray/Execution - APPEALS-43404](https://jira.devops.va.gov/secure/XrayExecuteTest!default.jspa?testExecIssueKey=APPEALS-43404&testIssueKey=APPEALS-43400)

# Best practices
## Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [x] No new code climate issues added

### CODE REVIEW

File: app/services/events/decision_review_created/decision_review_created_parser.rb [hunk 1]
```diff
@@ -7,8 +7,26 @@ class Events::DecisionReviewCreated::DecisionReviewCreatedParser
...
+  # methods to load example response
+  class << self
+    def example_response
+      File.read("#{Rails.root}/app/services/events/decision_review_created/decision_review_created_example.json")
+    end
+
+    def load_example
+      sample_header = {
+        "X-VA-Vet-SSN" => "123456789",
+        "X-VA-File-Number" => "123456789",
```
Comment (nkutub) [inline app/services/events/decision_review_created/decision_review_created_parser.rb:19]: I would make this a different number than SSN so that rspec will catch if we are parsing the correct field

File: app/services/events/decision_review_created/decision_review_created_parser.rb [hunk 2]
```diff
@@ -7,8 +7,26 @@ class Events::DecisionReviewCreated::DecisionReviewCreatedParser
 
   attr_reader :headers, :payload
 
-  def initialize(headers, payload)
-    @payload = payload
+  # methods to load example response
+  class << self
+    def example_response
+      File.read("#{Rails.root}/app/services/events/decision_review_created/decision_review_created_example.json")
```
Comment (nkutub) [inline app/services/events/decision_review_created/decision_review_created_parser.rb:13]: use the comma separated method you used in the rspec, its cleaner and easier to reed.

File: app/services/events/decision_review_created/decision_review_created_example.json [hunk 3]
```diff
@@ -0,0 +1,83 @@
+{
+  "css_id": "BVADWISE",
```
Comment (nkutub) [inline app/services/events/decision_review_created/decision_review_created_example.json:2]: I just want to make sure that these values are fictional and not a real vet's information

Comment (Jruuuu) [reply app/services/events/decision_review_created/decision_review_created_example.json:2]: yes sir
i believe eli gave these info as mock up and not actual vet info

File: app/services/events/decision_review_created/decision_review_created_parser.rb (hunk 2 above)
Comment (Jruuuu) [reply app/services/events/decision_review_created/decision_review_created_parser.rb:13]: Yes Sir, 

Updated

File: app/services/events/decision_review_created/decision_review_created_parser.rb (hunk 1 above)
Comment (Jruuuu) [reply app/services/events/decision_review_created/decision_review_created_parser.rb:19]: Yes sir
updated

File: spec/services/events/create_veteran_on_event_spec.rb [hunk 4]
```diff
@@ -61,14 +61,15 @@ def get_headers
     end
 
     def get_payload
-      {
+      data = {
```
Comment (jtsangVA) [inline spec/services/events/create_veteran_on_event_spec.rb:64]: Why was this change needed?

Comment (Jruuuu) [reply spec/services/events/create_veteran_on_event_spec.rb:64]: we refactored the parser to accept the JSON file and turn on the symbolize_names.

`   def initialize(headers, payload_json)
    @payload = JSON.parse(payload_json, symbolize_names: true)
    @headers = headers
    @veteran = @payload.dig(:veteran)
  end`
  
  in your test it was a ruby hash, i had to make the change so the get_payload would be in JSON

**Review by nkutub — APPROVED** [review]
Great work
